motionplan/tpSpaceRRT.go

```diff
@@ -391,64 +387,84 @@ func (mp *tpSpaceRRTMotionPlanner) getExtensionCandidate(
 	if err != nil || bestNode == nil {
 		return nil, err
 	}
-	pose, err := curPtg.Transform(bestNode.Configuration)
-	if err != nil {
-		return nil, err
-	}
-
-	bestDist := targetFunc(&ik.State{Position: pose})
-	goalAlpha := bestNode.Configuration[0].Value
-	goalD := bestNode.Configuration[1].Value
-
-	// Check collisions along this traj and get the longest distance viable
-	trajK, err := curPtg.Trajectory(goalAlpha, goalD)
-	if err != nil {
-		return nil, err
+	bestNodes := []node{}
+	pose := spatialmath.NewZeroPose()
+	// We may produce more than one consecutive arc. Reduce the one configuration to several 2dof arcs
+	for i := 0; i < len(bestNode.Configuration); i += 2 {
+		subNode := newConfigurationNode(bestNode.Configuration[i : i+2])
+		if invert {
+			bestNodes = append([]node{subNode}, bestNodes...)
+		} else {
+			bestNodes = append(bestNodes, subNode)
+		}
 	}
-	finalTrajNode := trajK[len(trajK)-1]
 
 	arcStartPose := nearest.Pose()
-	if invert {
-		arcStartPose = spatialmath.PoseBetweenInverse(finalTrajNode.Pose, arcStartPose)
-	}
-
-	sinceLastCollideCheck := 0.
-	lastDist := 0.
-	var nodePose spatialmath.Pose
-	// Check each point along the trajectory to confirm constraints are met
-	for i := 0; i < len(trajK); i++ {
-		trajPt := trajK[i]
+	successNodes := []node{}
+NODECHECK:
+	for _, subNode := range bestNodes {
```

Comment:
100% agree on combining the loops.

This can't really be framed as a Segment and processed properly under our current architecture. Completing https://viam.atlassian.net/browse/RSDK-5007 would make that easy though; I'll add a TODO naming that ticket.

Reply:
Awesome thanks!